packages/terra-table/src/SelectableTableRows.jsx

```diff
@@ -31,24 +35,60 @@ const propTypes = {
 const defaultProps = {
   disableUnselectedRows: false,
   onChange: undefined,
+  onBlur: undefined,
   selectedIndexes: [],
 };
 
 class SelectableTableRows extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
   }
 
+
   handleOnChange(event, index) {
     if (this.props.onChange) {
       this.props.onChange(event, index);
     }
+
+    if (!document.activeElement.hasAttribute('aria-label')) {
+      // Aria-Label is being populated with the cell contents for screen reader to read it as whole instead of cell value.
+      const ariaLabel = 'row selected';
+      document.activeElement.setAttribute('aria-label', ariaLabel);
+    }
+  }
+
+  handleOnBlur(event, index) {
+    if (this.props.onBlur) {
+      this.props.onBlur(event, index);
+    }
+  }
+
+  wrappedOnBlur(row, index) {
+    const initialOnBlur = row.props.onBlur;
+    // removes aria-label from unselected row
+    return (event) => {
+      if (row.props.isSelectable !== false) {
+        this.handleOnBlur(event, index);
+      }
+
+      const rowGroup = document.querySelector('[data-terra-row-group]');
```

Comment:
This affects the first table with this data attribute on the page and there is no guarantee that it is operating on the table that we want it to operate upon.

Reply:
yes..! found that while testing with multiple tables it picks the first row group always. fixed this by moving code to handleOnBlur which provides event parameter which holds the selected row as value for 'currentTarget' Property. so that current row can be retrieved by referring to event.currentTarget : [code](https://github.com/cerner/terra-core/blob/cc6ee3fad6bea01cc24bb5ce812e6483291d781d/packages/terra-table/src/SelectableTableRows.jsx#L81)